docs/concepts/cluster-administration/cloud-providers.md

```diff
@@ -56,6 +56,33 @@ Different settings can be applied to a load balancer service in AWS using _annot
 
 The information for the annotations for AWS is taken from the comments on [aws.go](https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/aws/aws.go)
 
+# OPENSTACK
+This section describes all the possible configurations which can
+be used while setting Kubernetes as an OpenStack provider.
+
+## cloud.conf
+Kubernetes configuration is modified via the file cloud.conf.It is the file that will tell Kubernetes where are your credentials and OpenStack auth endpoint.
+You can create a cloud.conf file by specifying the following details in it
+
+### Minimal configuration
+This is an example of a minimal configuration that touches the values that most often need to be set:
+```yaml
+[Global]
+username=user  
+password=pass  
+auth-url=https://<openstack_endpoint>:5000/v3  
+tenant-id=c869168a828847f39f7f06edd7305637  
+domain-id=2a73b8f597c04551a0fdc8e95544be8a
+
+[LoadBalancer]
+subnet-id=6937f8fa-858d-4bc9-a3a5-18d2c957166a  
+```
+
+*`username and password`: speak for themselves
+*`auth-url`: Used to specify the keystone endpoint.Can be found at : Access and Security > API Access > Credentials
+*`tenant-id`: Used to specify the id of the project where you want to create your resources.
+*`domain-id`: Used to specify the id of the domain your user belongs to.
+*`subnet-id`: Used to specify the id of the subnet you want to create your loadbalancer on. Can be found at : Network > Networks and click on the respective network to get its subnets
 {% endcapture %}
```

Comment:
Can we document all the config values [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/openstack/openstack.go#L114-L131) while we're at it? So we'll need:

`user-id`
`tenant-name`
`trust-id` (I had no idea what this was until I read [this](https://wiki.openstack.org/wiki/Keystone/Trusts#TrustID))
`domain-name`
`region`
`ca-file`

Reply:
@ss22ever Did you add these? These are global config options, not LB (which we'll do in a separate PR)

Reply:
@jamiehannaford I thought of adding it on a separate pr